### PR TITLE
[innogysmarthome] NullPointerException fixed which could occur when no device state is available (#9660)

### PR DESCRIPTION
**Pull request by Novanic**
This pull request fixes issue #9660, please see issue #9660 for more information.

The pull request contains also some refactorings (mainly to reduce duplicate code) and fixes potential bugs on setting the device reachable and low battery attributes.

### CODE REVIEW

File: bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/manager/FullDeviceManager.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
```
Comment (cpmeister) [inline bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/manager/FullDeviceManager.java:2]:
```suggestion
 * Copyright (c) 2010-2021 Contributors to the openHAB project
```

File: bundles/org.openhab.binding.innogysmarthome/src/test/java/org/openhab/binding/innogysmarthome/internal/client/entity/device/DeviceTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
```
Comment (cpmeister) [inline bundles/org.openhab.binding.innogysmarthome/src/test/java/org/openhab/binding/innogysmarthome/internal/client/entity/device/DeviceTest.java:2]:
```suggestion
 * Copyright (c) 2010-2021 Contributors to the openHAB project
```

**Review by cpmeister — CHANGES_REQUESTED** [review]
Other than the copyright changes, LGTM.

**Comment by Novanic** [comment]
@cpmeister When will I learn to not "simply" apply commit suggestions? ... :-( DCO failed and the suggested method to repair it (with a rebase) destroys the history. Than I have to create a new branch and a new pull request, just for changing 2020 to 2021 in 2 files... :-(

**Comment by fwolter** [comment]
The rebasing command, suggested by the DCO check doesn't work, because you merged the main branch into your branch several times.

Try `git rebase origin~2 --signoff`

Better use rebase instead of merge in PRs.

**Comment by Novanic** [comment]
Pull request broken...

**Comment by Novanic** [comment]
> The rebasing command, suggested by the DCO check doesn't work, because you merged the main branch into your branch several times.
> 
> Try `git rebase origin~2 --signoff`
> 
> Better use rebase instead of merge in PRs.

Hi, thank you for the hint. I tried that, but again everything is broken. Now the pull request has over 100 reviewers....... :-( I try your hint with rebasing instead of merging next time. Thank you.

**Comment by fwolter** [comment]
I'm wondering what causes these problems. Can you post the output of `git remote --verbose`?

**Comment by fwolter** [comment]
Sorry, it was my fault. It should have been simply `git rebase HEAD~2 --signoff`.

**Comment by Novanic** [comment]
New pull request: https://github.com/openhab/openhab-addons/pull/9677

**Pull request closed without merging**